polar-core/src/vm.rs

```diff
@@ -1962,8 +1962,9 @@ impl PolarVirtualMachine {
         let right = &args[1];
         let result = &args[2];
 
-        if !matches!(result.value(), Value::Variable(_)) {
-            return invalid_state(format!("arithmetic_op_helper: not a variable: {}", result));
+        match result.value() {
```

Comment:
Same possible replacement here? Make this `result.as_symbol()?`

Reply:
7589343e

nice nit!